### ISSUE TITLE
NFT: Question, why is the deposit not transferred to the new owner when buying an NFT?

### ISSUE THREAD

**Issue by SCJangra** [paritytech/polkadot-sdk#2851]
Hi, so when we set some metadata for an NFT we reserve some funds depending on the length of the metadata, and the same happens for setting attributes.

When someone buys an NFT they pay whatever price the seller has set. But I find it strange that even after selling an NFT, the original owner still has to pay the metadata and attribute deposit. It makes sense to me that when someone sells an NFT their reserved tokens are freed and the same amount of buyer's tokens are reserved. I am wondering why that is not the case.

Thanks

**Comment by bkchr** [comment]
CC @jsidorenko

**Comment by jsidorenko** [comment]
Hmm.. actually, it's a good idea! :)
Initially, we were moving the deposit onto a new owner on each NFT transfer but then removed that feature since you could create and send thousands of NFTs and lock funds on any account. Having to transfer the deposit on swap/sale only makes more sense!

**Comment by SCJangra** [comment]
Yes, it doesn't make sense to move deposit when someone is transferring the NFT, but for trades deposit should be moved. And for transfers, in addition to the current functionality, there could be additional functions that use a flow similar to what is being used to transfer ownership of collections.

**Comment by bkchr** [comment]
@SCJangra would you like to fix this?

**Comment by SCJangra** [comment]
@bkchr I would like to work on it.

**Comment by bkchr** [comment]
Perfect!
